packages/sanes-chrome-extension/src/extension/messages.ts

```diff
@@ -52,3 +56,29 @@ export async function sendCreatePersonaMessage(mnemonic?: string): Promise<Creat
     });
   });
 }
+
+export type GetPersonaResponse = PersonaData | null;
+
+export async function sendGetPersonaMessage(): Promise<GetPersonaResponse> {
+  return new Promise((resolve, reject) => {
+    const message: MessageToBackground = { action: MessageToBackgroundAction.GetPersona };
+    chrome.runtime.sendMessage(message, response => {
+      const lastError = chrome.runtime.lastError;
+      if (lastError) {
+        const errorMessage = lastError.message || 'Unknown error in chrome.runtime.lastError';
+        reject(errorMessage);
+        return;
+      }
+
+      if (!response) {
```

Comment:
I would do:
```
const persona = response ? { ...response } : null
resolve(persona);
```

Reply:
Since `response` is of type any, we should compose the `GetPersonaResponse` explicitly to make sure to copy the data we want. I tried to use only one resolve but the code does not become nicer than the `if`

```
      resolve(
        response
          ? {
              accounts: response.accounts,
              mnemonic: response.mnemonic,
              txs: response.txs,
            }
          : null
      );
```

Reply:
And what about:
```
const persona = response ? { ...(response as Whatever) } : null
resolve(persona);
```

Again, I do not see anything wrong in your solution, just wanted to improve readability. Your decision!

Reply:
Casting with `as` only changes type information at compile time. At runtime, any garbage from `response` will be copied. So in this case I don't see a nicer way of writing it than the current version.

Reply:
I see. Sounds good then.